Sources/App/Controllers/Registration/RegistrationController.swift

```diff
@@ -0,0 +1,84 @@
+import HTTP
+import Vapor
+import Fluent
+
+final class  RegistrationController {
+  
+  let autoapprove = try? AutoapproveController()
+  
+  func store(_ req: Request) throws -> ResponseRepresentable {
+    
+    guard
+      let token = req.headers.userToken,
+      let session = try Session.find(by: token),
+      let userId = session.user?.id
+    else {
```

Comment:
This guard with 3 lines is used in many places. I think it could be refactored to something like this
>    guard let userId = req.user?.id else {

Reply:
And in all places where it's used

Reply:
Moved it into helper function 
```swift
func getUserId(from req: Request)
```